pkg/subnet/local_test.go

```diff
@@ -180,32 +217,18 @@ func getTestClientFunc() (client.Client, error) {
 	c.On("RemoveSnapshot", mock.Anything, mock.Anything).Return(fakeRemoveSnapshotResponse, nil)
 	c.On("CreateBlockchains", mock.Anything, mock.Anything, mock.Anything).Return(fakeCreateBlockchainsResponse, nil)
 	c.On("URIs", mock.Anything).Return([]string{"fakeUri"}, nil)
-	fakeHealthResponse := &rpcpb.HealthResponse{
-		ClusterInfo: &rpcpb.ClusterInfo{
-			Healthy:          true, // currently actually not checked, should it, if CustomVMsHealthy already is?
-			CustomVmsHealthy: true,
-			NodeInfos: map[string]*rpcpb.NodeInfo{
-				"testNode1": {
-					Name: "testNode1",
-					Uri:  "http://fake.localhost:12345",
-				},
-				"testNode2": {
-					Name: "testNode2",
-					Uri:  "http://fake.localhost:12345",
-				},
-			},
-			CustomVms: map[string]*rpcpb.CustomVmInfo{
-				"vm1": {
-					BlockchainId: "abcd",
-				},
-				"vm2": {
-					BlockchainId: "efgh",
-				},
-			},
-			Subnets: []string{"subnet1", "subnet2"},
-		},
-	}
-	c.On("Health", mock.Anything).Return(fakeHealthResponse, nil)
+	// When fake deploying, the first response needs to have a bogus subnet ID, because
+	// otherwise the doDeploy function "aborts" when checking if the subnet had already been deployed.
+	// Afterwards, we can set the actual VM ID so that the test returns an expected subnet ID...
+
+	// Return a fake health response twice
+	c.On("Health", mock.Anything).Return(fakeHealthResponse, nil).Twice()
+	// Afterwards, change the VmId so that TestDeployToLocal has the correct ID to check
+	alteredFakeResponse := proto.Clone(fakeHealthResponse).(*rpcpb.HealthResponse) // new(rpcpb.HealthResponse)
+	alteredFakeResponse.ClusterInfo.CustomVms["vm2"].VmId = testVMID
+	alteredFakeResponse.ClusterInfo.CustomVms["vm2"].VmName = "test"
```

Comment:
Why is this `test` and not `vm2`? Maybe we should make these constants?

Reply:
Made changes to reflect this fact, so keys are now `bchainX`